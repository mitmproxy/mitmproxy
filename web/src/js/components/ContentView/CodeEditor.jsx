import React, { Component, PropTypes } from 'react'
import { render } from 'react-dom';
import Codemirror from 'react-codemirror';


CodeEditor.propTypes = {
        content: PropTypes.string.isRequired,
        onChange: PropTypes.func.isRequired,
}

export default function CodeEditor ( { content, onChange} ){

    let options = {
        lineNumbers: true
    };
    return (
        <div className="codeeditor" onKeyDown={e => e.stopPropagation()}>
            <Codemirror value={content} onChange={onChange} options={options}/>
        </div>
    )
}
