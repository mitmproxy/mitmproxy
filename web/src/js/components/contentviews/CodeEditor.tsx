import * as React from "react";
import { Component } from "react";
import CodeMirror from "../../contrib/CodeMirror";

type CodeEditorProps = {
    initialContent: string;
};

export default class CodeEditor extends Component<CodeEditorProps> {
    private editor = React.createRef<CodeMirror>();

    getContent = (): string => {
        return this.editor.current?.codeMirror.getValue();
    };

    render = () => {
        const options = {
            lineNumbers: true,
        };
        return (
            <div className="codeeditor" onKeyDown={(e) => e.stopPropagation()}>
                <CodeMirror
                    ref={this.editor}
                    value={this.props.initialContent}
                    onChange={() => 0}
                    options={options}
                />
            </div>
        );
    };
}
