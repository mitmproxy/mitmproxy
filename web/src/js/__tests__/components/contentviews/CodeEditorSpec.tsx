jest.mock("../../../contrib/CodeMirror");
import * as React from "react";
import CodeEditor from "../../../components/contentviews/CodeEditor";
import { render } from "../../test-utils";

test("CodeEditor", async () => {
    const { asFragment } = render(<CodeEditor initialContent="foo" />);
    expect(asFragment()).toMatchSnapshot();
});
