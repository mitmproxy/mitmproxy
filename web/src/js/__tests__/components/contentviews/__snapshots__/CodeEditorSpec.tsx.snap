// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor 1`] = `
<DocumentFragment>
  <div
    class="codeeditor"
  >
    <div
      class="cm-theme-light"
    >
      <div
        class="cm-editor ͼ1 ͼ2 ͼ4 ͼ17 ͼ15"
      >
        <div
          aria-live="polite"
          class="cm-announced"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              >
                1
              </div>
            </div>
            <div
              class="cm-gutter cm-foldGutter"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                <span
                  title="Unfold line"
                >
                  ›
                </span>
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              />
            </div>
          </div>
          <div
            aria-multiline="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4;"
            translate="no"
            writingsuggestions="false"
          >
            <div
              class="cm-activeLine cm-line"
            >
              foo
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CodeEditor highlights CSS 1`] = `
<DocumentFragment>
  <div
    class="codeeditor"
  >
    <div
      class="cm-theme-light"
    >
      <div
        class="cm-editor ͼ1 ͼ2 ͼ4 ͼ1g ͼ15"
      >
        <div
          aria-live="polite"
          class="cm-announced"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              >
                1
              </div>
            </div>
            <div
              class="cm-gutter cm-foldGutter"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                <span
                  title="Unfold line"
                >
                  ›
                </span>
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              />
            </div>
          </div>
          <div
            aria-multiline="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            data-language="css"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4;"
            translate="no"
            writingsuggestions="false"
          >
            <div
              class="cm-activeLine cm-line"
            >
              * { color: 
              <span
                class="ͼc"
              >
                black
              </span>
              ; }
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CodeEditor highlights HTML 1`] = `
<DocumentFragment>
  <div
    class="codeeditor"
  >
    <div
      class="cm-theme-light"
    >
      <div
        class="cm-editor ͼ1 ͼ2 ͼ4 ͼ1d ͼ15"
      >
        <div
          aria-live="polite"
          class="cm-announced"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              >
                1
              </div>
            </div>
            <div
              class="cm-gutter cm-foldGutter"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                <span
                  title="Unfold line"
                >
                  ›
                </span>
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              />
            </div>
          </div>
          <div
            aria-multiline="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            data-language="html"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4;"
            translate="no"
            writingsuggestions="false"
          >
            <div
              class="cm-activeLine cm-line"
            >
              &lt;
              <span
                class="ͼi"
              >
                p
              </span>
              &gt;yay&lt;/
              <span
                class="ͼi"
              >
                p
              </span>
              &gt;
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CodeEditor highlights JavaScript 1`] = `
<DocumentFragment>
  <div
    class="codeeditor"
  >
    <div
      class="cm-theme-light"
    >
      <div
        class="cm-editor ͼ1 ͼ2 ͼ4 ͼ1a ͼ15"
      >
        <div
          aria-live="polite"
          class="cm-announced"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              >
                1
              </div>
            </div>
            <div
              class="cm-gutter cm-foldGutter"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                <span
                  title="Unfold line"
                >
                  ›
                </span>
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              />
            </div>
          </div>
          <div
            aria-multiline="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            data-language="javascript"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4;"
            translate="no"
            writingsuggestions="false"
          >
            <div
              class="cm-activeLine cm-line"
            >
              alert(
              <span
                class="ͼd"
              >
                1
              </span>
              );
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CodeEditor highlights YAML 1`] = `
<DocumentFragment>
  <div
    class="codeeditor"
  >
    <div
      class="cm-theme-light"
    >
      <div
        class="cm-editor ͼ1 ͼ2 ͼ4 ͼ1j ͼ15"
      >
        <div
          aria-live="polite"
          class="cm-announced"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              >
                1
              </div>
            </div>
            <div
              class="cm-gutter cm-foldGutter"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                <span
                  title="Unfold line"
                >
                  ›
                </span>
              </div>
              <div
                class="cm-gutterElement cm-activeLineGutter"
                style="height: 14px;"
              />
            </div>
          </div>
          <div
            aria-multiline="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            data-language="yaml"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4;"
            translate="no"
            writingsuggestions="false"
          >
            <div
              class="cm-activeLine cm-line"
            >
              <span
                class="ͼl"
              >
                foo
              </span>
              : bar
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
